typescript/test/bitcoin.test.ts

```diff
@@ -464,4 +465,122 @@ describe("Bitcoin", () => {
       expect(targetToDifficulty(target)).to.equal(expectedDifficulty)
     })
   })
+
+  describe("createOutputScriptFromAddress", () => {
+    context("with testnet addresses", () => {
+      const btcAddresses = {
+        P2PKH: {
+          address: "mjc2zGWypwpNyDi4ZxGbBNnUA84bfgiwYc",
+          redeemerOutputScript:
+            "0x1976a9142cd680318747b720d67bf4246eb7403b476adb3488ac",
+          outputScript: "76a9142cd680318747b720d67bf4246eb7403b476adb3488ac",
+        },
+        P2WPKH: {
+          address: "tb1qumuaw3exkxdhtut0u85latkqfz4ylgwstkdzsx",
+          redeemerOutputScript:
+            "0x160014e6f9d74726b19b75f16fe1e9feaec048aa4fa1d0",
+          outputScript: "0014e6f9d74726b19b75f16fe1e9feaec048aa4fa1d0",
+        },
+        P2SH: {
+          address: "2MsM67NLa71fHvTUBqNENW15P68nHB2vVXb",
+          redeemerOutputScript:
+            "0x17a914011beb6fb8499e075a57027fb0a58384f2d3f78487",
+          outputScript: "a914011beb6fb8499e075a57027fb0a58384f2d3f78487",
+        },
+        P2WSH: {
+          address:
+            "tb1qau95mxzh2249aa3y8exx76ltc2sq0e7kw8hj04936rdcmnynhswqqz02vv",
+          redeemerOutputScript:
+            "0x220020ef0b4d985752aa5ef6243e4c6f6bebc2a007e7d671ef27d4b1d0db8dcc93bc1c",
+          outputScript:
+            "0020ef0b4d985752aa5ef6243e4c6f6bebc2a007e7d671ef27d4b1d0db8dcc93bc1c",
+        },
+      }
+
+      Object.entries(btcAddresses).forEach(
+        ([
+          addressType,
+          {
+            address,
+            redeemerOutputScript: expectedRedeemerOutputScript,
+            outputScript: expectedOutputScript,
+          },
+        ]) => {
+          it(`should create correct output script for ${addressType} address type`, () => {
+            const result = createOutputScriptFromAddress(address)
+
+            expect(result.toString()).to.eq(expectedOutputScript)
+            // Check if we can build the prefixed raw redeemer output script based
+            // on the result.
+            expect(buildRawPrefixedOutputScript(result.toString())).to.eq(
+              expectedRedeemerOutputScript
+            )
```

Comment:
Isn't it guaranteed if the previous assertion passes? 

`buildRawPrefixedOutputScript` is only converting the string to a buffer and prepends the length. If the `result` is `expectedOutputScript` this will always work. I would consider removing this assertion, the most important part of this test is whether `createOutputScriptFromAddress` recognizes and handles the given address type correctly.

Reply:
b9633d1